docs/plugins.md

```diff
@@ -0,0 +1,352 @@
+# Plugin Handbook
+
+## Table of Contents
+
+1. [Introduction](#introduction)
+   1. [What is a Lighthouse Plugin?](#what-is-a-lighthouse-plugin)
+   1. [Plugins v. Custom Config](#plugins-v-custom-config)
+   1. [Getting Started](#getting-started)
+1. [API](#api)
+   1. [Plugin Config](#plugin-config)
+   1. [Plugin Audits](#plugin-audits)
+1. [Best Practices](#best-practices)
+   1. [Naming](#naming)
+   1. [Scoring](#scoring)
+   1. [Common Mistakes](#common-mistakes)
+1. [Examples](#examples)
+
+## Introduction
+
+If you're new to Lighthouse development, start by reading up on the overall [architecture](./architecture.md) and how [configuration](./configuration.md) works before continuing.
+
+### What is a Lighthouse Plugin?
+
+Lighthouse plugins are a way to extend the functionality of Lighthouse with insight from domain experts (that's you!) and easily share this extra functionality with other Lighthouse users. At its core, a plugin is a node module that implements a set of checks that will get run by Lighthouse and added to the report as a new category.
+
+### Plugins v. Custom Config
+
+Plugins are easily shared and have a stable API that won't change between minor version bumps but are also more limited in scope. You might need a [custom Lighthouse configuration](./configuration.md) instead. Before getting started with plugins, think about your current needs, and consult the table below to decide which is best for you.
+
+| Capability                           | Plugin | Custom Config |
+| ------------------------------------ | ------ | ------------- |
+| Add custom audits                    | ✅      | ✅            |
+| Add custom categories                | ✅      | ✅            |
+| Easily sharable via NPM              | ✅      | ❌            |
```

Comment:
you'd just need an absolute path to the config in an npm module, but sure :)

Reply:
yeah sharable via NPM felt slightly off, what I want to get at here is that custom configs aren't sustainably sharable. you can only ever use one, they can't extend each other, etc. plugins are the real sharable way to offer bits of lighthouse logic.

better name/phrase to capture that complex thought? :)